### PR TITLE
Create modify-harness-delegate-kubernetes-api-client-timeout.md

### PR DESCRIPTION
**Pull request by abhinavsingh1196**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [x] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* https://harness.atlassian.net/browse/CDS-73125
* https://harness.atlassian.net/browse/CDS-71328

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

**Comment by abhinavsingh1196** [comment]
https://64997fe43be35a08dd1b1eed--harness-developer.netlify.app/docs/continuous-delivery/deploy-srv-diff-platforms/kubernetes/cd-kubernetes-category/modify-harness-delegate-kubernetes-api-client-timeout

@roshnisarangadharan please review

**Comment by roshnisarangadharan** [comment]
@abhinavsingh1196 This is a FirstGen item and cannot be documented in NextGen docs. We don't update FirstGen feature doc anymore. I've added this to the FirstGen SaaS Release Notes, https://github.com/harness/developer-hub/pull/2256. I've added the details you added here in that respective ticket. Thanks! I'll close this PR once your confirm.

File: docs/continuous-delivery/deploy-srv-diff-platforms/kubernetes/cd-kubernetes-category/modify-harness-delegate-kubernetes-api-client-timeout.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
+title: Modify Harness delegate's Kubernetes API client timeout
+description: Modify the read and connection timeouts for Kubernetes API client within Harness delegate
```
Comment (roshnisarangadharan) [inline docs/continuous-delivery/deploy-srv-diff-platforms/kubernetes/cd-kubernetes-category/modify-harness-delegate-kubernetes-api-client-timeout.md:3]:
```suggestion
description: Modify the read and connection timeouts for Kubernetes API client within Harness Delegate.
```

File: docs/continuous-delivery/deploy-srv-diff-platforms/kubernetes/cd-kubernetes-category/modify-harness-delegate-kubernetes-api-client-timeout.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+sidebar_position: 10
+---
+
+
+Harness delegate uses Kubernetes java client to make programmatic API calls to the provided Kubernetes server. For example, some common API calls include various CRUD calls for Kubernetes objects like ConfigMaps, Secrets etc.
```
Comment (roshnisarangadharan) [inline docs/continuous-delivery/deploy-srv-diff-platforms/kubernetes/cd-kubernetes-category/modify-harness-delegate-kubernetes-api-client-timeout.md:8]:
```suggestion
Harness Delegate uses Kubernetes Java client to make programmatic API calls to the Kubernetes server. For example, some common API calls include various REST API calls for CRUD operations for Kubernetes objects like ConfigMaps, Secrets, and so on.
```

File: docs/continuous-delivery/deploy-srv-diff-platforms/kubernetes/cd-kubernetes-category/modify-harness-delegate-kubernetes-api-client-timeout.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+
+## Configuration
+
+The API client uses a okhttp client whose default [read timeout](https://square.github.io/okhttp/4.x/okhttp/okhttp3/-ok-http-client/-builder/read-timeout/) and [connect timeout](https://square.github.io/okhttp/4.x/okhttp/okhttp3/-ok-http-client/-builder/connect-timeout/) values are set to 120 and 60 seconds respectively.
```
Comment (roshnisarangadharan) [inline docs/continuous-delivery/deploy-srv-diff-platforms/kubernetes/cd-kubernetes-category/modify-harness-delegate-kubernetes-api-client-timeout.md:13]:
```suggestion
The API client uses an OkHttp client whose default [read timeout](https://square.github.io/okhttp/4.x/okhttp/okhttp3/-ok-http-client/-builder/read-timeout/) and [connect timeout](https://square.github.io/okhttp/4.x/okhttp/okhttp3/-ok-http-client/-builder/connect-timeout/) values are set to 120 and 60 seconds respectively.
```

File: docs/continuous-delivery/deploy-srv-diff-platforms/kubernetes/cd-kubernetes-category/modify-harness-delegate-kubernetes-api-client-timeout.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+These values can be configured using environment variables, modifying the delegate's container environment. Values should be specified in SECONDS. 
+
+The environment variables for these timeouts are:
+
+ - Read timeout: `K8S_API_CLIENT_READ_TIMEOUT`
```
Comment (roshnisarangadharan) [inline docs/continuous-delivery/deploy-srv-diff-platforms/kubernetes/cd-kubernetes-category/modify-harness-delegate-kubernetes-api-client-timeout.md:19]:
```suggestion
 - Read timeout: `K8S_API_CLIENT_READ_TIMEOUT`.
```

File: docs/continuous-delivery/deploy-srv-diff-platforms/kubernetes/cd-kubernetes-category/modify-harness-delegate-kubernetes-api-client-timeout.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+ - Connect timeout: `K8S_API_CLIENT_CONNECT_TIMEOUT`
```
Comment (roshnisarangadharan) [inline docs/continuous-delivery/deploy-srv-diff-platforms/kubernetes/cd-kubernetes-category/modify-harness-delegate-kubernetes-api-client-timeout.md:20]:
```suggestion
 - Connect timeout: `K8S_API_CLIENT_CONNECT_TIMEOUT`.
```

**Review by roshnisarangadharan — DISMISSED** [review]
Approved after modifying the doc. Let me know when to merge.

**Comment by abhinavsingh1196** [comment]
@roshnisarangadharan looks good to merge

**Review by roshnisarangadharan — APPROVED** [review]

**Pull request merged**